### PR TITLE
Remove Custom Resource Definition management

### PR DESCRIPTION
**Pull request by dgrisonnet**
Changes:
- Removed CRD management from the operators
- Removed related options:
  - manage-crds
  - crd-kinds
  - with-validation
- Removed `crd-kinds` package
- Moved some CRD helper functions to `e2e/framework/crd.go`
- Manually create CRDs in e2e tests
- Removed test verifying that the operator was creating the CRDs
- Removed `bindata` usage

cc @lilic @brancz @pgier 

Any ideas where I should document this change?

### CODE REVIEW

**Comment by paulfantom** [comment]
We should also adjust RBAC policy

**Comment by dgrisonnet** [comment]
@paulfantom good catch! I updated the rbac rules

File: scripts/tooling/Dockerfile [hunk 1]
```diff
@@ -21,7 +20,6 @@ RUN GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@${
 RUN GO111MODULE=on go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb@${JB_VERSION}
 RUN go get github.com/brancz/gojsontoyaml
 RUN go get github.com/campoy/embedmd
-RUN GO111MODULE=on go get github.com/go-bindata/go-bindata/v3/go-bindata@${GO_BINDATA_VERSION}
```
Comment (paulfantom) [inline scripts/tooling/Dockerfile:24]: This might cause potential issues in other repositories depending on tooling image.

Comment (dgrisonnet) [reply scripts/tooling/Dockerfile:24]: Do we have an issue regarding the tooling migration where I could reference that?

**Comment by lilic** [comment]
lgtm letting @paulfantom have another look.

cc @coreos/team-monitoring PTAL as well.

File: scripts/tooling/Dockerfile (hunk 1 above)
Comment (paulfantom) [reply scripts/tooling/Dockerfile]: no, feel free to open one :)

**Comment by lilic** [comment]
@dgrisonnet do you mind rebasing and we can merge this, thanks!

**Comment by dgrisonnet** [comment]
Rebased :rocket:

**Pull request merged**